teacher.rb

```diff
@@ -0,0 +1,18 @@
+require_relative 'person'
+
+# class for teacher
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(age, specialization, name = 'unknown')
+    super(age, name)
+    @specialization = specialization
+  end
```

Comment:
- [ ] Again here kindly add the parent permission parameter to your initialize method as per requirement.

Code example
```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```